src/transformers/generation/flax_utils.py

```diff
@@ -954,7 +957,20 @@ def beam_search_body_fn(state, input_ids_length=1):
                 state.is_sent_finished.all(axis=-1, keepdims=True), did_topk_just_finished.shape
             ) & (early_stopping is True)
             add_penalty = ~did_topk_just_finished | beams_in_batch_are_full
-            topk_log_probs += add_penalty * np.array(-1.0e7)
+
+            # Add additional logic for diverse beam search
```

Comment:
Nice! My only nit is that we try and avoid `lambda` functions in `transformers` - would you be able to re-write these as standard function definitions please?

Reply:
Yes

Reply:
Lovely thanks